site_tools/image_deployer.py

```diff
@@ -0,0 +1,316 @@
+import argparse
+import datetime
+import json
+import logging
+import tempfile
+import yaml
+
+import openstack
+
+from itertools import islice
+
+from swiftclient.client import Connection as SwiftConnection
+
+
+logging.basicConfig(level=logging.INFO)
+
+
+class Image:
+    def __init__(self, name, base_container, scope, current_path):
+        self.name = name
+        self.manifest_name = name + ".manifest"
+        self.raw_name = name + ".raw"
+        self.qcow_name = name + ".qcow2"
+        self.scope = scope
+        self.base_container = base_container
+        self.current_path = current_path
+        self.container_path = self.base_container + "/" + self.current_path
+
+    def __str__(self):
+        return f"Image(name={self.name})"
+
+
+def get_openstack_connection(cloud, cloud_name):
+    return openstack.connect(auth_url=cloud['clouds'][cloud_name]["auth"]["auth_url"],
+                             auth_type="v3applicationcredential",
+                             application_credential_id=cloud['clouds'][cloud_name]["auth"]["application_credential_id"],
+                             application_credential_secret=cloud['clouds'][cloud_name]["auth"]["application_credential_secret"],
+                             region_name=cloud['clouds'][cloud_name]["region_name"])
```

Comment:
you can reference a clouds.yaml entry just by `openstack.connect(cloud="key_in_clouds_yaml")`
and it will pull in those other values

Reply:
done